_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -2264,7 +2264,7 @@ public static function validate_alphanum( $value = '', $request, $param ) {
 	 * @return bool|WP_Error
 	 */
 	public static function validate_verification_service( $value = '', $request, $param ) {
-		if ( ! empty( $value ) && ! ( is_string( $value ) && ( preg_match( '/^[a-z0-9_-]+$/i', $value ) || preg_match( '#^<meta name="([a-z0-9_\-.:]+)?" content="([a-z0-9_-]+)?" />$#i', $value ) ) ) ) {
+		if ( ! empty( $value ) && ! ( is_string( $value ) && ( preg_match( '/^[a-z0-9_-]+$/i', $value ) || preg_match( '#^<meta[^<>]+content=["\']?([^"\' <>]*)[^<>]*/?>$#i', $value ) ) ) ) {
```

Comment:
If the `content` attribute has an apostrophe or quotation mark in it, this regex will fail.

`<meta content="Jetpack's the best" />`

Reply:
It could also perceive this HTML as passing the check:

`<meta name="content=" />`

Reply:
Or even `<meta-content=>`.  Instead of duplicating work, this function should be calling `jetpack_verification_get_code()` and checking if it returns something, since that's the code that will actually be extracting the verification code.  (`jetpack_verification_get_code()` should also be given a second look, as its regex is pretty liberal as well.)